src/object.d

```diff
@@ -3251,11 +3252,31 @@ if (__traits(isFloating, F))
         if (!r) r = __cmp(f1.im, f2.im);
         return r;
     }
-    else static if (is(F == float) || is(F == double) || is(F == real))
+    else
     {
         return (f1 > f2) - (f2 > f1);
     }
-    else static assert(false, "Internal error");
```

Comment:
Why this change? Being explicit is generally better than hoping that the catch-all (else/default) does the right thing.